### PR TITLE
Only display properties with values for MeasureInfo

### PR DESCRIPTION
**Pull request by powercode**
## PR Summary
Adding Item selection conditions to the list formatting info for MeasureInfo so that only the properties with values gets displayed

Reopening #6214

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/powershell/7104)

### CODE REVIEW

**Comment by iSazonov** [comment]
From #6412 https://github.com/PowerShell/PowerShell/pull/6214#issuecomment-369424174
> @PowerShell/powershell-committeeTeam members are private reviewed this and is ok with the public apis

**Comment by iSazonov** [comment]
Reopen the PR to restart CIs.

**Comment by powercode** [comment]
I don't intend to fix the remaining CodeFactor issues.

**Comment by anmenaga** [comment]
@JamesWTruher can you please take a look? Thx.

**Comment by anmenaga** [comment]
Jim is on a leave; + @SteveL-MSFT for review who is familiar with formatting APIs.

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs [hunk 1]
```diff
@@ -370,7 +370,7 @@ internal ListEntryBuilder(ListControlBuilder listBuilder, ListControlEntry listE
             _listEntry = listEntry;
         }
 
-        private ListEntryBuilder AddItem(string value, string label, DisplayEntryValueType kind, string format)
+        private ListEntryBuilder AddItem(string value, string label, DisplayEntryValueType kind, string format, DisplayEntry itemSelectionContition)
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs]: typo: itemSelectionCondition

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs [hunk 2]
```diff
@@ -379,22 +379,30 @@ private ListEntryBuilder AddItem(string value, string label, DisplayEntryValueTy
...
+                FormatString = format,
+                ItemSelectionCondition = itemSelectionContition,
             });
 
             return this;
         }
 
         /// <summary></summary>
-        public ListEntryBuilder AddItemScriptBlock(string scriptBlock, string label = null, string format = null)
+        public ListEntryBuilder AddItemScriptBlock(string scriptBlock, string label = null, string format = null, DisplayEntry itemSelectionContition = null)
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs]: typo: itemSelectionCondition

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs [hunk 3]
```diff
@@ -379,22 +379,30 @@ private ListEntryBuilder AddItem(string value, string label, DisplayEntryValueTy
...
         {
-            return AddItem(scriptBlock, label, DisplayEntryValueType.ScriptBlock, format);
+            return AddItem(scriptBlock, label, DisplayEntryValueType.ScriptBlock, format, itemSelectionContition);
         }
 
         /// <summary></summary>
-        public ListEntryBuilder AddItemProperty(string property, string label = null, string format = null)
+        public ListEntryBuilder AddItemProperty(string property, string label = null, string format = null, DisplayEntry itemSelectionContition = null)
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs]: typo: itemSelectionCondition

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs [hunk 4]
```diff
@@ -379,22 +379,30 @@ private ListEntryBuilder AddItem(string value, string label, DisplayEntryValueTy
...
         {
-            return AddItem(property, label, DisplayEntryValueType.Property, format);
+            return AddItem(property, label, DisplayEntryValueType.Property, format, itemSelectionContition);
+        }
+
+        /// <summary></summary>
+        public ListEntryBuilder AddItemPropertyIfSet(string property, string label = null, string format = null)
+        {
+            var itemSelectionContition = DisplayEntry.CreatePropertyEntry(property);
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData_List.cs]: typo: itemSelectionCondition

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

File: test/powershell/Modules/Microsoft.PowerShell.Utility/object.tests.ps1 [hunk 5]
```diff
@@ -85,18 +85,30 @@ Describe "Object cmdlets" -Tags "CI" {
...
+        It 'should only display fields that are set' {
+            $text = 1, 2, 3 | Measure-Object -Sum -Average | Format-List | Out-String
+            $text -match "min|max" | Should -BeFalse
+            $text -match 'Sum' | Should -BeTrue
+            $text -match 'Average' | Should -BeTrue
+
+            $text = 1, 2, 3 | Measure-Object -Minimum -Maximum | Format-List | Out-String
+            $text -match "min" | Should -BeTrue
+            $text -match "max" | Should -BeTrue
+            $text -match 'Average' | Should -BeFalse
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/object.tests.ps1]: Seems we skip
```powershell
$text -match 'Sum' | Should -BeFalse
```

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/object.tests.ps1]: @powercode Please address the comment.

**Comment by iSazonov** [comment]
@BrucePay @lzybkr Please look the PR.

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData.cs [hunk 6]
```diff
@@ -788,7 +788,27 @@ public DisplayEntry(string value, DisplayEntryValueType type)
...
 
-        /// <summary/>
+        /// <summary>
+        /// Creates a DisplayEntry for the given scriptblock.
+        /// </summary>
+        /// <param name="scriptblock">The content of the scriptblock.</param>
+        /// <returns>A <see cref="DisplayEntry"/> for the <paramref name="scriptblock"/>.</returns>
+        public static DisplayEntry CreateScriptBlockEntry(string scriptblock)
+        {
+            return new DisplayEntry(scriptblock, DisplayEntryValueType.ScriptBlock);
```
Comment (lzybkr) [inline src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData.cs]: @PaulHigin should review this new public api for any security implications in creating script blocks that might circumvent the language mode.

**Comment by SteveL-MSFT** [comment]
@iSazonov we should revert this commit until security review of the public api has been completed

**Comment by iSazonov** [comment]
@SteveL-MSFT Opened #7754.

File: src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData.cs (hunk 6 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/FormatAndOutput/common/DisplayDatabase/displayDescriptionData.cs]: @PaulHigin reviewed this and told me he doesn't see a security issue.  @PaulHigin can you confirm?

**Pull request merged**